### PR TITLE
Added `Enumerable#in_slices_of`

### LINKED ISSUE #12844: `Array` form of `#each_slice`
`#in_groups_of` has an `Array` form, a block form, and an `Iterator` form:

```crystal
[1, 2, 3].in_groups_of(2)                           # => [[1, 2], [3, nil]]
[1, 2, 3].in_groups_of(2) { |e| p e if nil.in?(e) } # => [3, nil]
[1, 2, 3].each.in_groups_of(2).next                 # => [1, 2]
```

`#each_slice` only has the last two:

```crystal
[1, 2, 3].each_slice(2) { |e| p e if e.size == 1 } # => [3]
[1, 2, 3].each.slice(2).next                       # => [1, 2]
[1, 2, 3].each_slice(2).next                       # => [1, 2] # `Iterable` alias
```

There should be a method to fulfil that missing form, say:

```crystal
module Enumerable(T)
  def in_slices_of(count : Int) : Array(Array(T))
  end
end

[1, 2, 3].in_slices_of(2) # => [[1, 2], [3]]
```

### PR DESCRIPTION
**Pull request by pricelessrabbit**
Adding missing method and related tests

Resolves #12844

### CODE REVIEW

File: src/enumerable.cr [hunk 1]
```diff
@@ -629,6 +629,21 @@ module Enumerable(T)
     end
   end
 
+  # Returns an `Array` with chunks in the given size.
+  # Last chunk can be smaller depending on the number of remaining items
+  # [1, 2, 3].in_groups_of(2, 0) # => [[1, 2], [3, 0]]
```
Comment (straight-shoota) [inline src/enumerable.cr:634]: This doesn't work.
```suggestion
```

**Review by straight-shoota — APPROVED** [review]

File: src/enumerable.cr (hunk 1 above)
Comment (PricelessRabbit) [reply src/enumerable.cr]: sorry, fixed

File: src/enumerable.cr [hunk 2]
```diff
@@ -629,6 +629,20 @@ module Enumerable(T)
     end
   end
 
+  # Returns an `Array` with chunks in the given size.
+  # Last chunk can be smaller depending on the number of remaining items
+  # [1, 2, 3].in_slices_of(2)    # => [[1, 2], [3]]
+  # ```
```
Comment (Sija) [inline src/enumerable.cr:635]:
```suggestion
  # Returns an `Array` with chunks in the given size.
  # Last chunk can be smaller depending on the number of remaining items.
  #
  # ```
  # [1, 2, 3].in_slices_of(2) # => [[1, 2], [3]]
  # ```
```

**Review by Sija — APPROVED** [review]

File: spec/std/enumerable_spec.cr [hunk 3]
```diff
@@ -569,6 +569,19 @@ describe "Enumerable" do
     end
   end
 
+  describe "in slices of" do
+    it { [1, 2, 3].in_slices_of(1).should eq([[1], [2], [3]]) }
+    it { [1, 2, 3].in_slices_of(2).should eq([[1, 2], [3]]) }
+    it { [1, 2, 3, 4].in_slices_of(3).should eq([[1, 2, 3], [4]]) }
+    it { ([] of Int32).in_slices_of(2).should eq([] of Array(Array(Int32))) }
```
[the remark this answers is not in the document]
Comment (straight-shoota) [reply spec/std/enumerable_spec.cr:576]: This makes me wish for an equality expectation that also takes type into account...

[the remark this answers is not in the document]
Comment (PricelessRabbit) [reply spec/std/enumerable_spec.cr:576]: I'm missing the point about the wrong type when array is empty. shouldn't it be the same as cases where some item is present?

[the remark this answers is not in the document]
Comment (asterite) [reply spec/std/enumerable_spec.cr:576]: `.should be_empty`?

[the remark this answers is not in the document]
Comment (HertzDevil) [reply spec/std/enumerable_spec.cr:576]: `[] of Array(Array(Int32))` is a three-dimensional structure, whereas the method should only return a two-dimensional structure.

[the remark this answers is not in the document]
Comment (PricelessRabbit) [reply spec/std/enumerable_spec.cr:576]: Right! type is wrong also fot other methods like `#in_groups_of` and this misled me.

So how i should proceed?

* keep the `[] of Array(Array(Int32))` form or change to `.should be_empty`?
* can i search and fix also tests of other method in this PR?

[the remark this answers is not in the document]
Comment (asterite) [reply spec/std/enumerable_spec.cr:576]: I don't think making something about this is important. I suggest we merge this and maybe capture a different issue to unify the code.

[the remark this answers is not in the document]
Comment (straight-shoota) [reply spec/std/enumerable_spec.cr:576]: I created a new issue to continue the discussion about stricter type checking for equality expectations: https://github.com/crystal-lang/crystal/issues/13389

For this PR I think it's best to apply the suggested change to `[] of Array(Int32)`. Then this should be ready to merge.

**Review by straight-shoota — APPROVED** [review]

**Review by HertzDevil — APPROVED** [review]

**Pull request merged**